### PR TITLE
Fix duration parser and add tests

### PR DESCRIPTION
**Pull request by SphericalKat**
- Extract the parser to a separate function
 - Use UTC `DateTime` from the `chrono` crate instead of `SystemTime`
 - Handle an edge case where division by zero could occur
 - Add tests for parser

### CODE REVIEW

File: hcctl/src/main.rs [hunk 1]
```diff
@@ -146,3 +140,41 @@ fn list(settings: Settings) -> Result<()> {
...
+    use chrono::TimeZone;
+
+    use super::*;
+
+    #[test]
+    fn duration_parses_correctly() {
+        let now = &Utc.ymd(2021, 1, 26).and_hms(19, 38, 0);
+        let duration =
+            human_readable_duration(now, &"2021-01-26T14:00:24+00:00".to_owned()).unwrap();
+        assert!(duration == "5 hour(s) and 2 minute(s) ago")
```
Comment (msfjarvis) [inline hcctl/src/main.rs:170]: Does `assert_eq` not work for this? It has a better error message when the assertion fails in the comparison case.

**Review by msfjarvis — APPROVED** [review]
